### PR TITLE
Hardware config

### PR DESCRIPTION
**Pull request by GarrettBurroughs**
Created separate classes for hardware and control classes so physical input/hardware changes can be easily changed in the code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Power Up Branch: I2 [![Build Status](https://travis-ci.org/MontclairRobotics/PowerUp.svg?branch=i2)](https://travis-ci.org/MontclairRobotics/PowerUp)
 
+This is the second implementation of GradleRIO and the test of the IntelliJ platorm on Power Up Code for 2018
```
Comment (rafibaum) [inline README.md]: This is no longer true

File: src/main/java/frc/team555/robot/AutoModes.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import org.montclairrobotics.sprocket.auto.AutoMode;
+import org.montclairrobotics.sprocket.states.State;
+
+import java.util.ArrayList;
+
+public class AutoModes {
+    public static SmartDashboard dashboard;
+
+    public static AutoMode buildDashboardAuto(){
```
Comment (rafibaum) [inline src/main/java/frc/team555/robot/AutoModes.java]: What's this method supposed to do? What's the code in here supposed to accomplish?

File: src/main/java/frc/team555/robot/Hardware.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    //Drive Train Motor IDS
+    public static final int motorDriveBRID = 1;
+    public static final int motorDriveBLID = 2;
+    public static final int motorDriveFRID = 3;
+    public static final int motorDriveFLID = 4;
+
+
+    // Gyro ID
+    public static final SPI.Port navxPort = SPI.Port.kOnboardCS0;
```
Comment (rafibaum) [inline src/main/java/frc/team555/robot/Hardware.java]: Isn't the navX in kMXP?

File: src/main/java/frc/team555/robot/MecanumRobot.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
+package frc.team555.robot;
```
Comment (rafibaum) [inline src/main/java/frc/team555/robot/MecanumRobot.java]: Can't this be deleted as we're not doing mecanum?

File: src/main/java/frc/team555/robot/PowerUpRobot.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public void robotInit(){
+        Hardware.init();
+        DriveModule[] modules = new DriveModule[4];
+        modules[0] = new DriveModule(new XY(-1, -1), Vector.ZERO, new Motor(Hardware.motorDriveBL));
+        modules[1] = new DriveModule(new XY(1, -1), Vector.ZERO, new Motor(Hardware.motorDriveBR));
+        modules[2] = new DriveModule(new XY(-1, 1), Vector.ZERO, new Motor(Hardware.motorDriveFL));
+        modules[3] = new DriveModule(new XY(1, 1), Vector.ZERO, new Motor(Hardware.motorDriveFL));
+
+        driveTrain = new DriveTrain(modules);
```
Comment (rafibaum) [inline src/main/java/frc/team555/robot/PowerUpRobot.java]: Be sure you're instantiating properly with the constructor (I'm not sure if it's correct or not, will require testing).

**Review by rafibaum — CHANGES_REQUESTED** [review]
The .gradle files need to be removed and .gitignore'd

File: src/main/java/frc/team555/robot/AutoModes.java (hunk 2 above)
Comment (GarrettBurroughs) [reply src/main/java/frc/team555/robot/AutoModes.java]: Oh sorry, I didn't mean to have that, it was an accident

File: src/main/java/frc/team555/robot/PowerUpRobot.java (hunk 5 above)
Comment (GarrettBurroughs) [reply src/main/java/frc/team555/robot/PowerUpRobot.java]: Okay, ill test when we have the robot

File: src/main/java/frc/team555/robot/MecanumRobot.java (hunk 4 above)
Comment (GarrettBurroughs) [reply src/main/java/frc/team555/robot/MecanumRobot.java]: Yea, that was just left in there from when i branched off, ill get rid of it

File: src/main/java/frc/team555/robot/Hardware.java (hunk 3 above)
Comment (H1ppx) [reply src/main/java/frc/team555/robot/Hardware.java]: it should be SPI.Port.kMXP

File: src/main/java/frc/team555/robot/MecanumRobot.java (hunk 4 above)
Comment (H1ppx) [reply src/main/java/frc/team555/robot/MecanumRobot.java]: why is i2 code in this

File: src/main/java/frc/team555/robot/Hardware.java (hunk 3 above)
Comment (GarrettBurroughs) [reply src/main/java/frc/team555/robot/Hardware.java]: I just had that in there so i wouldn't get an error and planned on changing it later. I'll change it now though

**Review by H1ppx — DISMISSED** [review]
get rid of i2 stuff, pull direct from branch called steamworks instead, its more up to date and its tested

File: README.md (hunk 1 above)
Comment (H1ppx) [reply README.md]: @rafibaum he left i2 stuff in there for some reason

File: src/main/java/frc/team555/robot/Hardware.java (hunk 3 above)
Comment (H1ppx) [reply src/main/java/frc/team555/robot/Hardware.java]: it doesn't give an error tho

**Comment by GarrettBurroughs** [comment]
@H1ppx  I merged master into Hardware config and removed anything left over from i2, I didn't plan on merging until this is tested on the drivetrain next week once it is done.

**Comment by rafibaum** [comment]
Tested to fail. Please check code on RoboRIO.

**Comment by GarrettBurroughs** [comment]
Successful deploy, once tested on the actual robot should be ready for merging

**Comment by rafibaum** [comment]
Tested on the robot. All working except all inputs are reversed with identical motor inverts to spreadsheet. Some research needed.

**Review by rafibaum — APPROVED** [review]

**Pull request merged**